Orm/Xtensive.Orm/Orm/Linq/Expressions/EntityExpression.cs

```diff
@@ -19,94 +19,115 @@ internal class EntityExpression : ParameterizedExpression,
   {
     private List<PersistentFieldExpression> fields;
 
-    public TypeInfo PersistentType { get; private set; }
+    public TypeInfo PersistentType { get; }
 
-    public KeyExpression Key { get; private set; }
+    public KeyExpression Key { get; }
 
     public List<PersistentFieldExpression> Fields
     {
-      get { return fields; }
-      private set
-      {
+      get => fields;
+      private set {
         fields = value;
-        var fieldExpressions = fields.OfType<FieldExpression>();
-        foreach (var fieldExpression in fieldExpressions)
+        foreach (var fieldExpression in fields.OfType<FieldExpression>()) {
           fieldExpression.Owner = this;
+        }
       }
     }
 
     public bool IsNullable { get; set; }
 
     public Expression Remap(int offset, Dictionary<Expression, Expression> processedExpressions)
     {
-      if (!CanRemap)
+      if (!CanRemap) {
         return this;
-      Expression value;
-      if (processedExpressions.TryGetValue(this, out value))
+      }
+
+      if (processedExpressions.TryGetValue(this, out var value)) {
         return value;
+      }
 
       var keyExpression = (KeyExpression) Key.Remap(offset, processedExpressions);
       var result = new EntityExpression(PersistentType, keyExpression, OuterParameter, DefaultIfEmpty);
       processedExpressions.Add(this, result);
       result.IsNullable = IsNullable;
-      result.Fields = Fields
-        .Select(f => f.Remap(offset, processedExpressions))
-        .Cast<PersistentFieldExpression>()
-        .ToList();
+      var processedFields = new List<PersistentFieldExpression>(fields.Count);
+      foreach (var field in fields) {
+        // Do not convert to LINQ. We want to avoid a closure creation here.
+        processedFields.Add((PersistentFieldExpression) field.Remap(offset, processedExpressions));
+      }
+
+      result.Fields = processedFields;
       return result;
     }
 
     public Expression Remap(int[] map, Dictionary<Expression, Expression> processedExpressions)
     {
-      if (!CanRemap)
+      if (!CanRemap) {
         return this;
-      Expression value;
-      if (processedExpressions.TryGetValue(this, out value))
+      }
+
+      if (processedExpressions.TryGetValue(this, out var value)) {
         return value;
+      }
 
       var keyExpression = (KeyExpression) Key.Remap(map, processedExpressions);
-      if (keyExpression==null)
+      if (keyExpression==null) {

```

Comment:
According to the rules in ```.editorconfig``` file equality operators should be with spaces before and after